src/app/api/deps.py

```diff
@@ -48,37 +40,53 @@ async def get_current_access(security_scopes: SecurityScopes, token: str = Depen
 
     try:
         payload = jwt.decode(token, cfg.SECRET_KEY, algorithms=[cfg.JWT_ENCODING_ALGORITHM])
-        access_id = payload.get("sub")
-        if access_id is None:
-            raise unauthorized_exception("Invalid credentials", authenticate_value)
-        token_scopes = payload.get("scopes", [])
-        token_data = TokenPayload(access_id=int(access_id), scopes=token_scopes)
+    except JWTError:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Token has expired.",
+            headers={"WWW-Authenticate": authenticate_value},
+        )
 
-    except (JWTError, ValidationError, KeyError):
-        raise unauthorized_exception("Invalid credentials", authenticate_value)
-
-    entry = await crud.get(table=accesses, entry_id=int(access_id))
+    try:
+        access_id = int(payload["sub"])
+        token_scopes = payload.get("scopes", [])
+        token_data = TokenPayload(access_id=access_id, scopes=token_scopes)
+    except (KeyError, ValidationError):
+        raise HTTPException(
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            detail="Invalid token payload.",
+            headers={"WWW-Authenticate": authenticate_value},
+        )
```

Comment:
Don't you want to reuse unauthorized-exception with an additionnal argument for the status code ?

Reply:
The point of the PR is to get more informative / specific error messages and status code. Apart from the header, there isn't much in common with other exceptions that are thrown here :thinking: 

If it fails here, the token hasn't expired but for some reasons, its payload is invalid (missing entries in the dictionary most likely), so I figured it's best to throw a 422 error which is exactly made for this